ibind/support/oauth.py

```diff
@@ -0,0 +1,354 @@
+import base64
+import random
+import string
+import time
+from typing import Optional
+from urllib import parse
+
+from ibind import var
+from Crypto.Cipher import PKCS1_v1_5 as PKCS1_v1_5_Cipher
+from Crypto.Hash import SHA256, HMAC, SHA1
+from Crypto.PublicKey import RSA
+from Crypto.Signature import PKCS1_v1_5 as PKCS1_v1_5_Signature
+from cryptography.hazmat.primitives.serialization import load_pem_parameters
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:  # pragma: no cover
+    from ibind import IbkrClient
+
+
+def req_live_session_token(client: 'IbkrClient') -> tuple[str, int]:
+    """ Get live session token and access token from IBKR Web API used to make API endpoint calls """
+
+    ENCRYPTION_METHOD = "RSA-SHA256"
+    REQUEST_URL = "oauth/live_session_token"
+    REQUEST_BASE_URL = var.IBIND_OAUTH_REST_URL
+    REQUEST_METHOD = "POST"
+
+    prepend, extra_headers, dh_prime, dh_random = prepare_oauth()
+
+    headers = generate_oauth_headers(
+        request_method=REQUEST_METHOD,
+        request_url=f'{REQUEST_BASE_URL}{REQUEST_URL}',
+        signature_method=ENCRYPTION_METHOD,
+        extra_headers=extra_headers,
+        prepend=prepend        
+    )
+
+    # call ibind request
+    result = client.post(
+        path=REQUEST_URL,
+        base_url=REQUEST_BASE_URL, 
+        extra_headers=headers, 
+        log=True)
+
+    # TODO: catch error with result
+    # if result is not ok:
+    #     raise Exception(f"Live session token request failed: {response.text}")
```

Comment:
Do you need this addressed?

Reply:
I think any errors should be caught here, unless it's caught in rest_client\request?

Reply:
I see. Result won't tell us if it is not ok, instead an exception will be thrown. 

Here there's no way we could gracefully handle it so I'd recommend letting any exception from client.post just trickle up and remove this code.

Reply:
I'll remove this TODO